### PR TITLE
PR for BXMSDOC-2811: Document new kie-server persistence feature

### PR DESCRIPTION
**Pull request by kaldesai**
**JIRA:** https://issues.redhat.com/browse/BXMSDOC-2811

**Doc preview:**  [RHPAM-I. Managing and monitoring KIE Server](http://file.pnq.redhat.com/~kaldesai/BXMSDOC-2811-RHPAM-MMKS-QE/#proc-configuring-kie-server-persistence_execution-server)

**Doc impact:** Please check **18.1. Configuring KIE Server Persistence**

### CODE REVIEW

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 1]
```diff
@@ -0,0 +1,66 @@
+[id='proc-configuring-kie-server-persistence_{context}']
+
+= Configuring {KIE_SERVER} Persistence
+
+You can configure the {KIE_SERVER} persistence by passing hibernate or JPA parameters as system properties.
+
+To pass the hibernate or JPA parameters, use the following prefixes in your system properties:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:7]: Can we add these as prerequisites?

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+.Procedure
+
+. To configure {KIE_SERVER} persistence, complete any of the following tasks:
+
++
+--
+If you want to configure {KIE_SERVER} persistence using {EAP} configuration file, complete the following tasks:
+
+... In your {PRODUCT} installation directory, navigate to the `standalone-full.xml` file. For example, if you use {EAP} installation for {PRODUCT}, go to `$EAP_HOME/standalone/configuration/standalone-full.xml` file.
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:20]: I believe it should be second-level numbering

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+... Open the `standalone-full.xml` file and under the `<system-properties>` tag, set your hibernate or JPA parameters as system properties.
++
+The following example shows how to configure {KIE_SERVER} persistence using hibernate parameters:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:23]: I'd suggest adding this sentence as a tag to the source as follows:

.Example of configuring {KIE_SERVER} persistence using hibernate parameters
[source]
...

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+[source,xml,subs="attributes+"]
+----
+<system-properties>
+    ...
+        <property name="hibernate.hbm2ddl.auto" value="create-drop"/>
+    ...
+<system-properties>
+----
++
+The following example shows how to configure {KIE_SERVER} persistence using JPA parameters:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:34]: Same comment as above

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+----
+<system-properties>
+    ...
+        <property name="javax.persistence.jdbc.url" value="jdbc:mysql://mysql.db.server:3306/my_database?useSSL=false&serverTimezone=UTC"/>
+    ...
+<system-properties>
+----
+
+
+If you want to configure {KIE_SERVER} persistence using command line, complete the following tasks:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:46]: I'd suggest adding numbering or bullet to this point.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+* `javax.persistence`
+* `hibernate`
+
+.Procedure
+
+. To configure {KIE_SERVER} persistence, complete any of the following tasks:
+
++
+--
+If you want to configure {KIE_SERVER} persistence using {EAP} configuration file, complete the following tasks:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:18]: I'd suggest adding numbering or bullet to this point. I think bullet would be suitable.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+<system-properties>
+----
+
+
+If you want to configure {KIE_SERVER} persistence using command line, complete the following tasks:
+
+... Pass the parameters directly from the command line using `-Dkey=value` as follows:
+
++
+The following example shows how to configure {KIE_SERVER} persistence using hibernate parameters:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:51]: Same comment as above

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
++
+[source]
+----
+$EAP_HOME/bin/standalone.sh -Dhibernate.hbm2ddl.auto=create-drop
+----
++
+The following example shows how to configure {KIE_SERVER} persistence using JPA parameters:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:58]: same comment as above.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    ...
+        <property name="javax.persistence.jdbc.url" value="jdbc:mysql://mysql.db.server:3306/my_database?useSSL=false&serverTimezone=UTC"/>
+    ...
+<system-properties>
+----
+
+
+If you want to configure {KIE_SERVER} persistence using command line, complete the following tasks:
+
+... Pass the parameters directly from the command line using `-Dkey=value` as follows:
```
Comment (hmanwani-rh) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:48]: If there is only one step, then I believe numbering is not required. @emmurphy please confirm.

**Review by hmanwani-rh — COMMENTED** [review]
@kaldesai I've added a few suggestions below. Please let me know if you've any questions.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 2 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:20]: Done

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 3 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:23]: Updated

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 4 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:34]: Updated

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 6 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:18]: Updated. Added a bullet to this point.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 7 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:51]: Updated

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 8 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:58]: Updated

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 10]
```diff
@@ -0,0 +1,64 @@
+[id='proc-configuring-kie-server-persistence_{context}']
+
+= Configuring {KIE_SERVER} Persistence
+
+You can configure the {KIE_SERVER} persistence by passing hibernate or JPA parameters as system properties.
+
+To pass the hibernate or JPA parameters, use the following prefixes in your system properties:
```
Comment (MarianMacik) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:7]: I would rephrase this a bit, these prefixes are not prepended by a user, rather it means that properties with these prefixes are recognized by a KIE Server. So users can use every JPA or Hibernate parameters with these prefixes. Also see the original JIRA which says so: `kie-server will take properties with following prefix into consideration:`.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
++
+.Example of configuring {KIE_SERVER} persistence using JPA parameters:
++
+[source]
+----
+$EAP_HOME/bin/standalone.sh -Djavax.persistence.jdbc.url=jdbc:mysql://mysql.db.server:3306/my_database?useSSL=false&serverTimezone=UTC
+----
+
+--
+. Start the {KIE_SERVER}.
```
Comment (MarianMacik) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:64]: Those 2 commands will already start the KIE Server with the specified system properties.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+        <property name="javax.persistence.jdbc.url" value="jdbc:mysql://mysql.db.server:3306/my_database?useSSL=false&serverTimezone=UTC"/>
+    ...
+<system-properties>
+----
+
+If you want to configure {KIE_SERVER} persistence using command line, complete the following tasks:
+
+... Pass the parameters directly from the command line using `-Dkey=value` as follows:
++
+.Example of configuring {KIE_SERVER} persistence using hibernate parameters:
```
Comment (MarianMacik) [inline doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:49]: hibernate should by with a capital letter, i.e. Hibernate. It is a name of a community project. See https://hibernate.org/.

**Comment by emmurphy** [comment]
Please remove me from this email chain. I believe I was added as a typo.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 11 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc:64]: @MarianMacik, Okay, I will remove this step.

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 10 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc]: Updated

File: doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc (hunk 12 above)
Comment (kaldesai) [reply doc-content/enterprise-only/admin-and-config/proc-configuring-kie-server-persistence.adoc]: Updated and changed the word hibernate to Hibernate

**Review by MarianMacik — APPROVED** [review]
Looks good now, thanks!

**Comment by emmurphy** [comment]
Please remove me from your mailing list
